tests/test_utils.py

```diff
@@ -359,6 +360,30 @@ def test_cutline():
         with pytest.raises(RioTilerError):
             utils.create_cutline(src.dataset, bad_poly, geometry_crs="epsg:4326")
 
+    triangle_over_image_edge = {
+        "type": "Polygon",
+        "coordinates": [
+            [
+                [-104.775390888988852, 38.953714348778355],
+                [-104.775146720379681, 38.953580769848777],
+                [-104.775389629827075, 38.953472856486307],
+                [-104.775390888988852, 38.953714348778355],
+            ]
+        ],
+    }
+
+    # Check when using `boundless cutline`
+    # https://github.com/cogeotiff/rio-tiler/issues/585
+    triangle_bounds = featureBounds(triangle_over_image_edge)
+    with Reader(COG_RGB) as src:
+        cutline = utils.create_cutline(
+            src.dataset, triangle_over_image_edge, geometry_crs="epsg:4326"
+        )
+        data, mask = src.part(triangle_bounds, vrt_options={"cutline": cutline})
+        assert sum(mask[:, 0]) == 0  # first line
+        assert sum(mask[0, :]) == 0  # first column
+        assert sum(mask[:, -1]) == 0  # last column
```

Comment:
relying on pixel count is a bit tricky because it can changed depending on the platform and GDAL version. Those are more general tests to make sure it work with `boundless` cutline (first and last line should be only `0`

Reply:
Great, I did have doubts about the pixel count and like this solution a lot! Thanks 😄